server/container/guice/mailet/src/main/java/org/apache/james/modules/server/CamelMailetContainerModule.java

```diff
@@ -122,6 +125,7 @@ public MailetModuleConfigurationPerformer(ConfigurationProvider configurationPro
             this.mailQueueFactory = mailQueueFactory;
             this.transportProcessorCheckSet = transportProcessorCheckSet;
             this.defaultProcessorsConfigurationSupplier = defaultProcessorsConfigurationSupplier;
+            this.executorService = new SandglassExecutorService(Duration.ofMinutes(1));
```

Comment:
I'm sceptical with this *timeout* concept on the executor service. I'd rather say **let's wait they are all initiallized** rather than **check they are all well initialyzed after 1 minutes**. Looks like more robust to me...

Also, if you keep this timeout, it should be configurable.